src/librustc/traits/mod.rs

```diff
@@ -128,7 +135,8 @@ pub type TraitObligation<'tcx> = Obligation<'tcx, ty::PolyTraitPredicate<'tcx>>;
 static_assert_size!(PredicateObligation<'_>, 112);
 
 /// The reason why we incurred this obligation; used for error reporting.
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash, HashStable, RustcEncodable, RustcDecodable,
+         TypeFoldable)]
```

Comment:
Not that you can do much about it, but this looks like it'll add a lot of generated code to libsrutc -- making what is already the slowest-compiling crate even slower. Cc https://github.com/rust-lang/rust/issues/65031